telegram/_keyboardbuttonrequest.py

```diff
@@ -0,0 +1,282 @@
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser Public License for more details.
+#
+# You should have received a copy of the GNU Lesser Public License
+# along with this program.  If not, see [http://www.gnu.org/licenses/].
+"""
+This module contains two objects to request chats/users and their resulting service
+messages.
+"""
+from typing import TYPE_CHECKING, Optional
+
+from telegram._chatadministratorrights import ChatAdministratorRights
+from telegram._telegramobject import TelegramObject
+from telegram._utils.types import JSONDict
+
+if TYPE_CHECKING:
+    from telegram import Bot
+
+
+class KeyboardButtonRequestUser(TelegramObject):
+    """This object defines the criteria used to request a suitable user. The identifier of the
+    selected user will be shared with the bot when the corresponding button is pressed.
+
+    Objects of this class are comparable in terms of equality. Two objects of this class are
+    considered equal, if their :attr:`request_id` is equal.
+
+    .. versionadded:: 20.1
+
+    Args:
+        request_id (:obj:`int`): Signed 32-bit identifier of the request.
+        user_is_bot (:obj:`bool`, optional): Pass :obj:`True` to request a bot, pass :obj:`False`
+            to request a regular user. If not specified, no additional restrictions are applied.
+        user_is_premium (:obj:`bool`, optional): Pass :obj:`True` to request a premium user, pass
+            :obj:`False` to request a non-premium user. If not specified, no additional
+            restrictions are applied.
+    Attributes:
+        request_id (:obj:`int`): Identifier of the request.
+        user_is_bot (:obj:`bool`): Optional. Pass :obj:`True` to request a bot, pass :obj:`False`
+            to request a regular user. If not specified, no additional restrictions are applied.
+        user_is_premium (:obj:`bool`): Optional. Pass :obj:`True` to request a premium user, pass
+            :obj:`False` to request a non-premium user. If not specified, no additional
+            restrictions are applied.
+    """
+
+    __slots__ = (
+        "request_id",
+        "user_is_bot",
+        "user_is_premium",
+    )
+
+    def __init__(
+        self,
+        request_id: int,
+        user_is_bot: bool = None,
+        user_is_premium: bool = None,
+        *,
+        api_kwargs: JSONDict = None,  # skipcq: PYL-W0622
+    ):
+        super().__init__(api_kwargs=api_kwargs)
+        self.request_id: int = request_id
+
+        self.user_is_bot: Optional[bool] = user_is_bot
+        self.user_is_premium: Optional[bool] = user_is_premium
+
+        self._id_attrs = (self.request_id,)
+
+        self._freeze()
+
+
+class UserShared(TelegramObject):
```

Comment:
I like them together in one file.

Reply:
`*Shared` and `KeyboardButtonRequest*` have little in common imo. The former represent an event that is triggered by using the latter, but they are input for very different objects and share no common architecture …

Reply:
I still think they are very closely related which suffice having them in one file, but since harshil also thinks different file, different file it is.